### PR TITLE
more informative error message

### LINKED ISSUE #434: If can't connect to github to get the script -> show meaningful error
Today's GitHub outage revealed that we don't print anything if the script fails to fetch things from github. We should print `Couldn't connect to GitHub, please check your internet connection and githubstatus.com.`

```
kilrau@K-Yoga:~$ xud -b connext-yy (nothing printed)
kilrau@K-Yoga:~$ xud -b connext-yy (nothing printed)
kilrau@K-Yoga:~$ xud -b connext-yy (nothing printed)
kilrau@K-Yoga:~$ xud -b connext-yy (nothing printed)
kilrau@K-Yoga:~$ xud -b connext-yy
❌ Branch "connext-yy" does not exist (which is only half-true since branch exists, but we can't connect to github to find it) 
```

### PR DESCRIPTION
**Pull request by hatmer**
closes #434

### CODE REVIEW

**Comment by kilrau** [comment]
Can you try pushing the branch to this repo (`ExchangeUnion:github-error-if-github-is-down`)? Only this way our travis autobuild for docker images works unfortunately.

**Comment by raladev** [comment]
Addition to GitHub's shutdown.
First time github.com returned unicorn (500 error), but after some time github.com starts working, however https://api.github.com/repos/ExchangeUnion/xud-docker/git/refs/heads/$1 was still returning 500 error.
What do you think about additional checking, but without root of the github? @kilrau @hatmer 
Something like:
1. Try to get 200 for https://api.github.com/repos/ExchangeUnion/xud-docker/git/refs/heads/$1
2. If code is 404 then Branch does not exist
3. Other code: Something went wrong: got XXX response code from <URI>. Please check githubstatus.com
4. Timeout error: Couldn't connect to GitHub: please check your internet connection and githubstatus.com

**Comment by kilrau** [comment]
That sounds about right! @raladev

File: xud.sh [hunk 1]
```diff
@@ -19,7 +19,10 @@ function parse_arguments() {
                     exit 1
                 fi
                 if ! curl -sf -o /dev/null https://api.github.com/repos/ExchangeUnion/xud-docker/git/refs/heads/$1; then
-                    echo >&2 "❌ Branch \"$1\" does not exist"
+                    if ! curl -sL -w "%{http_code}" "https://www.github.com/" -o /dev/null | grep 200; then
```
Comment (reliveyy) [inline xud.sh:22]: How about use `grep -q 200` here to avoid strange line `200` in the screen when there is a connection issue with github?

**Comment by raladev** [comment]
@hatmer 
a. flow with timeout does not work for me and idk why -_-
1. To get 000, I used edited hosts file and it works (curl returns 000 in console)
2. But when I trying use xud.sh, i got nothing -_-
[Screenshot](https://user-images.githubusercontent.com/29906866/81164137-88bb5200-8f98-11ea-9e1a-a6c7deebc220.png)


@kilrau 
b. I found another flow with No output behavior, but it is not connected with status of github. Should we fix it in this PR or Should i create new issue?
1. https://api.github.com/repos/ExchangeUnion/xud-docker/git/refs/heads/$1 returns several records if $1 have no full match, for instance, for connext-y or github.
2. as a result u got no output in xud.sh for this case.
[Screenshot](https://user-images.githubusercontent.com/29906866/81164123-822cda80-8f98-11ea-9384-46b9a7c77387.png)

**Comment by kilrau** [comment]
Fix it here too

**Comment by hatmer** [comment]
Given that the git refs REST endpoints return multiple/partial matches for queries, I have modified the xud.sh logic to make just one curl request and then check the HTTP status of that request in order to provide helpful error messages in case of failure

**Comment by hatmer** [comment]
@raladev thank you for finding these edge cases!

**Pull request merged**